### PR TITLE
Incomplete Form tests

### PR DESCRIPTION
**Pull request by nsclark**
Tests for the form tab of the test plan.

Notes;

-Used language ccz for convenience when I started working on this. Can easily separate and change to another app if recommended, should just be a few changes of text in "" for first 2 scenarios. 
-The sync test section should pass, but I haven't been able to get I toggle airplane mode to succeed. Wanted to get into review while I worked on that.

### CODE REVIEW

**Comment by nsclark** [comment]
@phillipm 

Hey - let me know what you think. Tried to incorporate some of the feedback you gave from the language tests, so hopefully this is a little smoother as well. Still having issues with the toggle airplane mode command, so still looking into that portion of the test.

File: form_tests.feature [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	Then I press "Languages"
+	Then I press "Go To Start"
+	Then I see the text "test"
+	Then Submit
+
+	Then I see the text "Unsent Forms: 1"
+	Then I logout
+	Then I toggle airplane mode
+
+	Then I login with username "clark" and password "123"
```
Comment (phillipm) [inline form_tests.feature]: you probably don't need to logout and back in, just toggle airplane mode

File: form_tests.feature [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+	#Scenario covers Incomplete Forms 1 through 5
+	Then I install the ccz app at "languages.ccz"
+	Then I login with username "user_with_no_data" and password "123"
+	#Create Incomplete Form
+	Then I press start
+	Then I select module "Basic Form Tests"
+	Then I select module "Languages"
+	Then I wait
+	
+	Then I save form as incomplete
```
Comment (nsclark) [inline form_tests.feature]: Yup - added. Good catch.

File: form_tests.feature [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	Then I go back to the home screen
+
+Scenario: Save case forms
+	#Incomplete 7, Case Links 1 - 2b
+	Then I login with username "clark" and password "123"
+	Then I press start
+	Then I select module "Case Tests"
+	And I select module "Update a Case"
+
+	Then I select case "Snow"
```
Comment (nsclark) [inline form_tests.feature]: Forgot to add. Added to the main_structure_steps.rb in the same place as Saumya in his case tests.

File: form_tests.feature [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	Then I press the "Continue" button
+	Then I wait
+	Then I save form as incomplete
+	Then I wait to see "Start"
```
Comment (nsclark) [inline form_tests.feature]: Smart. Will do.

File: form_tests.feature [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+Scenario: Save case forms
+	#Incomplete 7, Case Links 1 - 2b
+	Then I login with username "clark" and password "123"
+	Then I press start
+	Then I select module "Case Tests"
+	And I select module "Update a Case"
+
+	Then I select case "Snow"
+	Then I press the "Continue" button
+	Then I wait
```
Comment (nsclark) [inline form_tests.feature]: Added to main structure steps

File: form_tests.feature [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+	Then I open incomplete forms
+	Then I press "Languages"
+	Then I press "Go To Start"
+
+	Then I see the text "test"
+	Then I go back to the home screen
+
+Scenario: Form Entry Quirks
+	# Form Entry Quirks 1
```
Comment (nsclark) [inline form_tests.feature]: There was an issue with being able to save a form that violated a constraint (particularly, when a response was required). Added comments to clarify

File: form_tests.feature [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	Then I select module "Case Tests"
+	And I select module "Update a Case"
+
+	Then I select case "Snow"
+	Then I press the "Continue" button
+	Then I wait to see "Continue Form"
+	Then I see the text "Delete old copy"
+	Then I see the text "No"
+	Then I see the text "Yes"
+	Then I press the "No" button
```
Comment (nsclark) [inline form_tests.feature]: Yup - can do!

File: form_tests.feature (hunk 4 above)
Comment (nsclark) [inline form_tests.feature]: I'm actually having a lot of trouble with this. I'm not sure how to modify the step definition to get this to work. Have tried several combos of wait_for_element_to_exist commands. I've added the step definition. Should the command be in the if portion or outside of it?

File: form_tests.feature (hunk 1 above)
Comment (nsclark) [inline form_tests.feature]: Yup, removing that actually got it working for me.

File: form_tests.feature [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+Scenario: Test incomplete form creation
+	#Using language.ccz instead of ccqa.ccz. Issues with accessing Basic Form of ccqa.ccz. Should only need to change a few lines/words in the first 2 scenarios of this test if I can bypass that issue. 
+	#Scenario covers Incomplete Forms 1 through 5
+	Then I install the ccz app at "languages.ccz"
+	Then I login with username "user_with_no_data" and password "123"
+	#Create Incomplete Form
+	Then I press start
+	Then I select module "Basic Form Tests"
+	Then I select module "Languages"
+	Then I wait
```
Comment (nsclark) [inline form_tests.feature]: Ignore previous comment. Got it to work with `I select form "Languages"`

**Pull request closed without merging**